microsetta_public_api/repo/_beta_repo.py

```diff
@@ -0,0 +1,39 @@
+from microsetta_public_api.repo._base import DiversityRepo
+from microsetta_public_api.exceptions import UnknownID, InvalidParameter
+import numpy as np
+
+
+class BetaRepo(DiversityRepo):
+
+    def exists(self, sample_ids, metric):
+        distance_matrix = self._get_resource(metric)
+        if isinstance(sample_ids, str):
+            return sample_ids in distance_matrix.ids
+        else:
+            existing_ids = set(distance_matrix.ids)
+            return [(id_ in existing_ids) for id_ in sample_ids]
+
+    def k_nearest(self, sample_id, metric, k=1):
+        distance_matrix = self._get_resource(metric)
+        if sample_id not in distance_matrix.ids:
```

Comment:
this should probably use the .exists method above on the assumption it moves to testing against a cached lookup

Reply:
https://github.com/biocore/microsetta-public-api/pull/68/commits/558430afaf593a1f7a2171ec7ee86e4f36a223eb